File: araok-authorization-service/src/main/java/ru/otus/dto/UserDto.java
package ru.otus.dto;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;
import ru.otus.domain.User;
import ru.otus.enums.RoleEnum;

import java.time.LocalDate;
import java.util.Collection;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserDto implements UserDetails {
    private Long id;

    private String name;

    private String phone;

    private String password;

    private LocalDate birthDate;

    private RoleEnum role;

    @Override
    @JsonIgnore
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return List.of(role);
    }

    @Override
    @JsonIgnore
    public String getUsername() {
        return String.valueOf(id);
    }

    @Override
    @JsonIgnore
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    @JsonIgnore
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    @JsonIgnore
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    @JsonIgnore
    public boolean isEnabled() {
        return true;
    }

    public static User toDomainObject(UserDto dto) {
        return User.builder()
                .id(dto.id)
                .name(dto.name)
                .phone(dto.phone)
                .password(dto.password)
                .birthDate(dto.birthDate)
                .role(dto.role)
                .build();
    }

    public static UserDto toDto(User user) {
        return UserDto.builder()
                .id(user.getId())
                .name(user.getName())
                .phone(user.getPhone())
                .password(user.getPassword())
                .birthDate(user.getBirthDate())
                .role(user.getRole())
                .build();
    }
}

File: araok-android/src/main/java/ru/otus/data/dto/UserDto.kt
package ru.araok.data.dto

import android.os.Build
import androidx.annotation.RequiresApi
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import ru.araok.entites.User
import java.time.LocalDate

@JsonClass(generateAdapter = true)
data class UserDto @RequiresApi(Build.VERSION_CODES.O) constructor(
    @Json(name = "id") override val id: Long = -1,
    @Json(name = "name") override val name: String = "",
    @Json(name = "phone") override val phone: String = "",
    @Json(name = "password") override val password: String = "",
    @Json(name = "birthDate") override val birthDate: LocalDate = LocalDate.now(),
    @Json(name = "role") override val role: String = ""
): User
